### PR TITLE
#164046214 Users can see list of articles

### PR DESCRIPTION
**Pull request by RachelleMaina**
### What does this PR do?
 -  Enable users to see list of articles in the application on the homepage
 -  add utils to get image from body and format db DateTime field
 -  styles homepage
 -  add utils tests

### Description
 -  add list article components, container, actions and actiontypes
 -  add utils to get image from body and format db DateTime field
 -  styles homepage
 -  add utils tests

### How has this been tested?
 - [x] Unit tests

### How to manually test this:
- Clone the repository `https://github.com/andela/ah-legion-frontend`
- Checkout the branch `ft-list-all-articles-164046214`
- Run `npm install` to install dependecies
-  Run `npm start' to fire up the dev server
-  Go to a browser and try visiting the home url ie. `/`

### PT
[#164046214](https://www.pivotaltracker.com/story/show/164046214)

### CODE REVIEW

**Comment by abulojoshua1** [comment]
@RachelleMaina, kindly fix the coverage drop.

**Comment by RachelleMaina** [comment]
> @RachelleMaina, kindly fix the coverage drop.

Will do this.

File: src/tests/axiosConfig.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import axios from 'axios';
```
Comment (MandelaK) [inline src/tests/axiosConfig.js]: I don't see any difference between this file and `src/axiosConfig.js`. Unless you have a compelling reason, I'd advise you delete this file and resort to importing from one module.

File: src/axiosConfig.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import axios from 'axios';
+
+const BASE_URL = 'https://ah-legion-staging.herokuapp.com/api/';
```
Comment (BrianSerem) [inline src/axiosConfig.js]: This URL should be in the .env file.

**Comment by RachelleMaina** [comment]
> @RachelleMaina, kindly fix the coverage drop.

It is all good now

**Review by MandelaK — COMMENTED** [review]
Just fix the linting errors, everything else looks good

File: src/containers/HomeView.jsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import { connect } from 'react-redux';
+import Home from '../components/Home';
+import { fetchAllArticles } from '../store/actions/articles';
+
+class HomeView extends Component {
+  componentDidMount() {
+    const { fetchAllArticles } = this.props;
+    fetchAllArticles()
+ 
+  }
```
[the remark this answers is not in the document]
Comment (abulojoshua1) [reply src/containers/HomeView.jsx:10]: Kindly fix this.

**Comment by abulojoshua1** [comment]
@RachelleMaina, kindly fix the HoundCi comments.

**Comment by RachelleMaina** [comment]
> @RachelleMaina, kindly fix the HoundCi comments.

I have fixed all of the comments except JsDoc Comments, I do not know why Hound is still capturing them.

**Review by Etomovich — APPROVED** [review]

**Review by ElMonstro — APPROVED** [review]
LGTM

**Review by BrianSerem — APPROVED** [review]

File: src/containers/HomeView.jsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+HomeView.propTypes = {
+  articles: PropTypes.arrayOf(PropTypes.oneOfType([PropTypes.number,
+    PropTypes.string, PropTypes.array])),
+
+};
+
+HomeView.defaultProps = {
+  articles: [],
+};
+export const mapStateToProps = state => ({ articles: state.articles });
```
Comment (MandelaK) [inline src/containers/HomeView.jsx]: Please destructure articles here 
```suggestion
export const mapStateToProps = state => ( const { articles } = state; { articles});
```

**Review by MandelaK — CHANGES_REQUESTED** [review]
Rachelle please untrack `package-lock.json` from this PR

File: src/containers/HomeView.jsx (hunk 4 above)
Comment (RachelleMaina) [reply src/containers/HomeView.jsx]: Will do this.

**Review by MandelaK — APPROVED** [review]

**Review by abulojoshua1 — APPROVED** [review]
LGTM

**Pull request merged**